src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs

```diff
@@ -118,6 +120,10 @@ protected void ValidateScopeInput(SortedSet<string> scopesToValidate)
                 RequestContext.Logger.Error(ex);
```

Comment:
I am confused by this, are you saying if RunAsync call ends up in the BaseRequest.cs (not overridden) then you will throw. But  you generate the requestId here?